src/IDCodeValidator.java

```diff
@@ -0,0 +1,175 @@
+            System.out.println(yearsOld + 1);
+            return true;
+        }
+        if ((!(yearsOld > 0))) {
+            System.out.println("Error");
+        }
+
+        return false;
+    }
+
+
+    public static boolean isLeapYear(int yearNumber) {
+        if ((yearNumber % 4 == 0 && yearNumber % 100 != 0) || (yearNumber % 400 == 0)) {
+            return true;
+        } else return false;
+    }
+
+
+    public static boolean checkControlNumber(String idCode) {
+        int result = 0;
+        List<Integer> a = new ArrayList<>();
+        a.add(Integer.valueOf(idCode));
+        for (int i = 0; i < idCode.length(); i++) {
+            String n = String.valueOf(idCode.charAt(i));
+            a.add(Integer.parseInt(n));
+        }
+        List<Integer> x = new ArrayList<>();
+        x.add(1);
+        x.add(2);
+        x.add(3);
+        x.add(4);
+        x.add(5);
+        x.add(6);
+        x.add(7);
+        x.add(8);
+        x.add(9);
+        x.add(1);
+        for (int i = 0; i < a.size() - 1; i++) {
+            result = result + (a.get(i) * x.get(i));
+        }
+        int leftover = result % 11;
+
+        if (leftover < 10) {
+            System.out.println(result);
+        }
+        if (leftover == 10) {
+            System.out.println(result == 0);
+        }
+        if (leftover > 10) {
+            System.out.println("Error");
+        }
+        return false;
+    }
+
+
+    public static String getInformation(String idCode) {
+        if (isCorrect(idCode)) {
+            IDNumberData personalCode = IDNumberData.getInformationFromIdCode(idCode);
+            int year = personalCode.yearNumber;
+            int genderNum = personalCode.genderNumber;
+            int month = personalCode.monthNumber;
+            int day = personalCode.dayNumber;
+            int fullYear = getFullYear(genderNum, year);
+            String gender = String.valueOf(getGender(genderNum));
+            return String.format("This is a " + gender + "born on " + convertIntToString(day), convertIntToString(month), fullYear);
+        } else {
+            return "ID is not correct";
+
+        }
+    }
+
+
+
+
+    public static Gender getGender(int genderNumber) {
+        if (genderNumber == 2 || genderNumber == 4 || genderNumber ==6) {
+            System.out.println("Gender number is correct");
+            return Gender.FEMALE;
+        }
+        else if (genderNumber == 1 || genderNumber == 3 || genderNumber == 5) {
+            System.out.println("Gender number is correct");
+            return Gender.MALE;
+        }else
+            throw new IllegalArgumentException("No such option");
+    }
+
+
+
+    public static int getFullYear(int genderNumber, int year) {
+        int years;
+        GregorianCalendar gcalendar = new GregorianCalendar();
+        System.out.println(years = gcalendar.get(Calendar.YEAR));
+        System.out.println(years);
+
+        return years;
+    }
```

Comment:
This is incorrect
```Java
System.out.println(IDCodeValidator.getFullYear(4, 98));
```
prints 2023
```java
System.out.println(IDCodeValidator.getFullYear(10, 98));
```
also prints 2023

Reply:
Please use here your method to validate that gender number is correct, create additional method to check that yearNumber is correct (year number can be any number from 0-99
Then do the calculatations
1 and 2 => 1800 + yearNumber
3 and 4 => 1900 + yearNumber
5 and 6 => 2000 + yearNumber